src/main/java/net/minecraftforge/gui/ForgeGuiHandler.java

```diff
@@ -0,0 +1,52 @@
+package net.minecraftforge.gui;
+
+import io.netty.channel.embedded.EmbeddedChannel;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.inventory.Container;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.FakePlayer;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.common.network.FMLOutboundHandler;
+import net.minecraftforge.fml.common.network.internal.FMLNetworkHandler;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.gui.capability.IGuiProvider;
+
+public class ForgeGuiHandler
+{
+    public static void openGui(EntityPlayer player, World worldObj, IGuiProvider provider)
+    {
+        if (provider == null) return;
+
+        Object owner = provider.getOwner(player, worldObj);
+        if (worldObj.isRemote)
+        {
+            Object gui = provider.getClientGuiElement(player, worldObj, owner);
+            FMLCommonHandler.instance().showGuiScreen(gui);
+            return;
+        }
+
+        if (player instanceof FakePlayer || player instanceof EntityPlayerSP) return;
```

Comment:
I think you should rearrange this method from here down to avoid early returns. 
Then you can have
```java
if (player instanceof EntityPlayerMP && !(player instanceof FakePlayer)) {
   ...
}
```